packages/strapi-plugin-content-manager/admin/src/containers/CollectionTypeWrapper/index.js

```diff
@@ -28,24 +29,41 @@ const CollectionTypeWrapper = ({ allLayoutData, children, from, slug }) => {
   const isCreatingEntry = id === 'create';
 
   const fetchURL = useMemo(() => {
-    if (isCreatingEntry) {
+    if (isCreatingEntry && !origin) {
       return null;
     }
 
-    return getRequestUrl(`${slug}/${id}`);
-  }, [slug, id, isCreatingEntry]);
+    return getRequestUrl(`${slug}/${origin || id}`);
+  }, [slug, id, isCreatingEntry, origin]);
+
+  const cleanClonedData = useCallback(
+    data => {
+      if (!origin) {
+        return data;
+      }
+      const cleaned = removeFieldsFromClonedData(
+        data,
+        allLayoutData.contentType,
+        allLayoutData.components
+      );
+
+      return cleaned;
+    },
+    [allLayoutData, origin]
+  );
 
   const cleanReceivedData = useCallback(
     data => {
+      console.log(cleanClonedData(data));
       const cleaned = removePasswordFieldsFromData(
-        data,
+        cleanClonedData(data),
```

Comment:
This should not be done here, but in the useEffect instead, directly when we fetch the data.

Reply:
`cleanReceivedData` is called many times in the file so I figured it would be better to do it there where there is already a cleaning for the passwords.
better everywhere data are called?

Reply:
Well IMO it only needs to be done when the data is fetched, after it's not necessary as the data is already formatted.